### PR TITLE
server: check finished Ok first when job is retried.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut  @disksing

### CODE REVIEW

File: server/cluster_worker.go [hunk 1]
```diff
@@ -173,23 +173,55 @@ func (c *raftCluster) updateJobStatus(job *pd_jobpd.Job, status pd_jobpd.JobStat
...
 		}
 	}
 
-	resp, err := c.processJob(job, firstRunning)
-	if err != nil || resp == nil {
-		return errors.Trace(err)
+	if resp == nil {
+		resp, err = c.processJob(job, checkOK)
+		if err != nil || resp == nil {
+			return errors.Trace(err)
```
Comment (disksing) [inline server/cluster_worker.go]: If both err and resp equals nil, we should return an error instead of nil.

Comment (siddontang) [inline server/cluster_worker.go]: no, if both nil, we can cancel this job safely outer.

Comment (disksing) [inline server/cluster_worker.go]: Got it. LGTM.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**